packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -34,6 +35,359 @@ enum DayPeriod {
+  /// fragment.
+  ///
+  /// This value contributes to the total horizontal width of all fragments
+  /// appearing on the same line, unless it is the left-most fragment on the
+  /// line, in which case this value is ignored.
+  final double leftMargin;
+}
+
+/// Displays the am/pm fragment and provides controls for switching between am
+/// and pm.
+class _DayPeriodControl extends StatelessWidget {
+  const _DayPeriodControl({
+    @required this.fragmentContext,
+  });
+
+  final _TimePickerFragmentContext fragmentContext;
+
+  void _handleChangeDayPeriod() {
+    final int newHour = (fragmentContext.selectedTime.hour + _kHoursPerPeriod) % _kHoursPerDay;
+    fragmentContext.onTimeChange(fragmentContext.selectedTime.replacing(hour: newHour));
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final TextTheme headerTextTheme = fragmentContext.headerTextTheme;
+    final TimeOfDay selectedTime = fragmentContext.selectedTime;
+    final Color activeColor = fragmentContext.activeColor;
+    final Color inactiveColor = fragmentContext.inactiveColor;
+
+    final TextStyle amStyle = headerTextTheme.subhead.copyWith(
+        color: selectedTime.period == DayPeriod.am ? activeColor: inactiveColor
+    );
+    final TextStyle pmStyle = headerTextTheme.subhead.copyWith(
+        color: selectedTime.period == DayPeriod.pm ? activeColor: inactiveColor
+    );
+
+    return new GestureDetector(
+      onTap: Feedback.wrapForTap(_handleChangeDayPeriod, context),
+      behavior: HitTestBehavior.opaque,
+      child: new Column(
+        mainAxisSize: MainAxisSize.min,
+        children: <Widget>[
+          new Text('AM', style: amStyle),
+          const SizedBox(width: 0.0, height: 4.0),  // Vertical spacer
+          new Text('PM', style: pmStyle),
+        ],
+      ),
+    );
+  }
+}
+
+/// Displays the hour fragment.
+///
+/// When tapped changes time picker dial mode to [_TimePickerMode.hour].
+class _HourControl extends StatelessWidget {
+  const _HourControl({
+    @required this.fragmentContext,
+    @required this.hourFormat,
+  });
+
+  final _TimePickerFragmentContext fragmentContext;
+  final _TimePickerHourFormat hourFormat;
+
+  @override
+  Widget build(BuildContext context) {
+    final TextStyle hourStyle = fragmentContext.mode == _TimePickerMode.hour
+      ? fragmentContext.activeStyle
+      : fragmentContext.inactiveStyle;
+
+    return new GestureDetector(
+      onTap: Feedback.wrapForTap(() => fragmentContext.onModeChange(_TimePickerMode.hour), context),
+      child: new Text(_formatHour(), style: hourStyle),
+    );
+  }
+
+  String _formatHour() {
+    switch(hourFormat) {
+      case _TimePickerHourFormat.h:
+        return fragmentContext.selectedTime.hourOfPeriodLabel;
+      case _TimePickerHourFormat.H:
+        return fragmentContext.selectedTime.hour.toString();
+      default:
+        return fragmentContext.selectedTime.hourLabel;
```

Comment:
I don't really understand what these three things mean

Reply:
oh never mind, i see.

I would move TimeOfDay to above all these classes, so it's clearer when you get here in the code what these properties mean.